hplsql/src/main/java/org/apache/hive/hplsql/functions/InMemoryFunctionRegistry.java

```diff
@@ -132,39 +133,84 @@ private boolean execProc(String name, HplsqlParser.Expr_func_paramsContext ctx)
   /**
    * Set parameters for user-defined function call
    */
-  public static void setCallParameters(String procName, HplsqlParser.Expr_func_paramsContext actual, ArrayList<Var> actualValues,
-                         HplsqlParser.Create_routine_paramsContext formal,
-                         HashMap<String, Var> out,
-                         Exec exec) {
-    if (actual == null || actual.func_param() == null || actualValues == null) {
+  public static void setCallParameters(String procName, HplsqlParser.Expr_func_paramsContext actual,
+      ArrayList<Var> actualValues, HplsqlParser.Create_routine_paramsContext formal, HashMap<String, Var> out,
+      Exec exec) {
+    // if it is a non-parameter function then just return.
+    if (actual == null && formal == null) {
       return;
     }
-    int actualCnt = actualValues.size();
-    int formalCnt = formal.create_routine_param_item().size();
-    if (formalCnt != actualCnt) {
-      throw new ArityException(actual.getParent(), procName, formalCnt, actualCnt);
+    int actualCnt = (actualValues == null) ? 0 : actualValues.size();
+    int passedParamCnt = actualCnt;
+    List<HplsqlParser.Create_routine_param_itemContext> routineParamItem = formal.create_routine_param_item();
+    int formalCnt = routineParamItem.size();
+    ParserRuleContext ruleContext = (actual == null) ? null : actual.getParent();
```

Comment:
Is it possible that `actualValues` is null and `actual` is not null or vice versa?

if not can we shortcut that code by checking whether any actual parameter passed and  extract the code part which handles the case when no actual parameter passed.

It would help get rid of conditions like `(actual == null)`

Reply:
Both `actualValues` and `actual` either will be null or non-null at the same time. Earlier if they are null they are just returning from `setCallParameters()` API but that was wrong as even procedure has params but user didn't pass anything it won't throw exception so that is corrected now. Also now we need to handle default parameters, if procedure has default params and user didn't pass any param then we should take default value for those params.

If there is any mismatch between procedure actual params and passed params by considering default params we throw exception, while throwing exception earlier it was passed `actual.getParent()` to the exception constructor, earlier if actual is null then just returning from method so actual can not be null if flow comes to throwing exception but now we need to handle default params also so in this case actual can be null so just added null check.